addons/project/views/project_task_views.xml

```diff
@@ -529,7 +530,8 @@
                     <group>
                         <field name="project_id" invisible="1" />
                         <field name="company_id" invisible="1" />
-                        <field name="parent_id" domain="[('id', '!=', id)]" context="{'search_default_project_id': project_id, 'search_default_open_tasks': 1}" /> 
+                        <field name="child_ids" invisible="1" />
+                        <field name="parent_id" domain="[('id', '!=', id), ('id', 'not in', child_ids), ('parent_id.id', 'not in', child_ids)]" context="{'search_default_project_id': project_id, 'search_default_open_tasks': 1}" />
```

Comment:
Why is the third part needed for the domain ? the ``('parent_id.id', 'not in', child_ids)``

Reply:
I have added this third part to restrict sub-sub-task, but now I have updated code and added:  '!', ('id', 'child_of', id)